clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1188,6 +1190,17 @@ private void close(long timeout, TimeUnit timeUnit, boolean swallowException) {
         }
     }
 
+    private static Map<String, Object> propsToMap(Properties properties) {
+        Map<String, Object> map = new HashMap<>(properties.size());
+        for (Object key : properties.keySet()) {
+            if (key instanceof String) {
```

Comment:
Currently `AbstractConfig(...)` constructor will throw `ConfigException(entry.getKey().toString(), entry.getValue(), "Key must be a string.")` if the key is not string. This can be useful to catch unintentionally mistake on the user side. Should we keep the same behavior in `propsToMap(...)`?

Reply:
Good catch. Fixed and added a test to verify.